### PR TITLE
[Merged by Bors] - feat: port GroupTheory.Subsemigroup.Center

### PR DESCRIPTION
**Pull request by LukasMias**

### CODE REVIEW

File: Mathlib/GroupTheory/Subsemigroup/Center.lean [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  { z | ∀ m, m * z = z * m }
+#align set.center Set.center
+#align set.add_center Set.addCenter
+
+-- porting note: The `to_additive` version used to be `mem_add_center` without the iff
+@[to_additive mem_addCenter_iff]
+theorem mem_center_iff [Mul M] {z : M} : z ∈ center M ↔ ∀ g, g * z = z * g :=
+  Iff.rfl
+#align set.mem_center_iff Set.mem_center_iff
+#align set.mem_add_center_iff Set.mem_addCenter_iff
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/Subsemigroup/Center.lean:47]:
```suggestion
#align set.mem_add_center Set.mem_addCenter_iff
```

File: Mathlib/GroupTheory/Subsemigroup/Center.lean [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+/-- The center of a semigroup `M` is the set of elements that commute with everything in `M` -/
+@[to_additive
+      "The center of a semigroup `M` is the set of elements that commute with everything in `M`"]
+def center : Subsemigroup M where
+  carrier := Set.center M
+  mul_mem' a b := Set.mul_mem_center a b
+#align subsemigroup.center Subsemigroup.center
+#align add_subsemigroup.center AddSubsemigroup.center
+
+-- porting note: `coe_center` is now redundant
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/Subsemigroup/Center.lean:157]:
```suggestion
-- porting note: `coe_center` is now redundant
#noalign subsemigroup.coe_center
#noalign addsubsemigroup.coe_center
```

**Review by Ruben-VandeVelde — APPROVED** [review]

File: lake-manifest.json [hunk 3]
```diff
@@ -1,57 +1,21 @@
 {"version": 4,
- "packagesDir": "./lake-packages",
+ "packagesDir": ".\\lake-packages",
```
Comment (hrmacbeth) [inline lake-manifest.json:2]: It was probably an accident to include your local changes to this manifest file.  Can you please revert them?

File: Mathlib/GroupTheory/Subsemigroup/Center.lean [hunk 4]
```diff
@@ -33,39 +33,46 @@ namespace Set
 variable (M)
 
 /-- The center of a magma. -/
-@[to_additive add_center " The center of an additive magma. "]
+@[to_additive addCenter " The center of an additive magma. "]
 def center [Mul M] : Set M :=
```
Comment (hrmacbeth) [inline Mathlib/GroupTheory/Subsemigroup/Center.lean:37]: I believe the [naming convention](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki#naming-convention) would be `Center`, `AddCenter`.

File: Mathlib/GroupTheory/Subsemigroup/Center.lean [hunk 5]
```diff
@@ -137,33 +147,35 @@ variable (M) [Semigroup M]
 
 /-- The center of a semigroup `M` is the set of elements that commute with everything in `M` -/
 @[to_additive
-      "The center of a semigroup `M` is the set of elements that commute with everything in\n`M`"]
+      "The center of a semigroup `M` is the set of elements that commute with everything in `M`"]
 def center : Subsemigroup M where
   carrier := Set.center M
-  mul_mem' a b := Set.mul_mem_center
+  mul_mem' a b := Set.mul_mem_center a b
```
Comment (hrmacbeth) [inline Mathlib/GroupTheory/Subsemigroup/Center.lean:153]: Does the following work?  (untested)
```suggestion
  mul_mem' := Set.mul_mem_center
```

File: lake-manifest.json (hunk 3 above)
Comment (LukasMias) [reply lake-manifest.json]: The linter was yelling at me for having an outdated lake manifest, so I pushed an updated one. Though I recognize leaving it unchanged is probably the best way. The manifest file has been acting weird for me the past couple days. Thanks!

File: Mathlib/GroupTheory/Subsemigroup/Center.lean (hunk 4 above)
Comment (LukasMias) [reply Mathlib/GroupTheory/Subsemigroup/Center.lean]: One of these days I'll get the naming conventions right from the get-go :) Thank you!

File: lake-manifest.json (hunk 3 above)
Comment (hrmacbeth) [reply lake-manifest.json]: CI is still yelling at you, I see.  Try merging master (`git merge origin/master`) and resolving any version changes in favour of what's on master, and see if that gets it to compile.

File: Mathlib/GroupTheory/Subsemigroup/Center.lean [hunk 6]
```diff
@@ -16,13 +16,13 @@ import Mathlib.GroupTheory.Subsemigroup.Operations
...
 ## Main definitions
 
-* `set.center`: the center of a magma
-* `subsemigroup.center`: the center of a semigroup
-* `set.add_center`: the center of an additive magma
-* `add_subsemigroup.center`: the center of an additive semigroup
+* `Set.Center`: the center of a magma
+* `Subsemigroup.Center`: the center of a semigroup
+* `Set.AddCenter`: the center of an additive magma
+* `addSubsemigroup.Center`: the center of an additive semigroup
```
Comment (hrmacbeth) [inline Mathlib/GroupTheory/Subsemigroup/Center.lean:22]:
```suggestion
* `AddSubsemigroup.Center`: the center of an additive semigroup
```

File: Mathlib/GroupTheory/Subsemigroup/Center.lean (hunk 4 above)
Comment (hrmacbeth) [reply Mathlib/GroupTheory/Subsemigroup/Center.lean]: Hah, blind leading the blind here ... :)

File: lake-manifest.json (hunk 3 above)
Comment (LukasMias) [reply lake-manifest.json]: ```git merge``` tells me that everything's up to date, and I'm seeing other PRs having the same issue (e.g. https://github.com/leanprover-community/mathlib4/pull/1277). Maybe the master manifest needs updating? :) I think I'll ask on the Zulip.

Comment (hrmacbeth) [reply lake-manifest.json]: I _think_ you can try merging master again once #1279 lands.

**Review by hrmacbeth — APPROVED** [review]

**Comment by dupuisf** [comment]
Thanks!

bors r+

**Pull request closed without merging**